### PR TITLE
Add Kingdom Hearts 2

### PR DESCRIPTION
**Pull request by shananas**
With collaberation with Ugh_Sunlight and refrence other Maguffin hunts yaml as thats all KH2 is.

### CODE REVIEW

**Comment by ScootyPuffJr1** [comment]
I've verified that this runs, but I don't know anything about the game to comment about option weights. Game weight seems ok.

File: games/Kingdom Hearts 2.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    'true': 50
+  LuckyEmblemsAmount:
+    30: 50
+    31: 50
+    32: 50
+    33: 50
+    34: 50
+    35: 50
+  LuckyEmblemsRequired:
+    20: 50
```
Comment (Silvris) [inline games/Kingdom Hearts 2.yaml:38]:
```suggestion
    random-range-20-25: 50
```

Comment (Silvris) [reply games/Kingdom Hearts 2.yaml:38]: Same here, can trim down a couple of rows

File: games/Kingdom Hearts 2.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    0: 50
+  Promise_Charm:
+    'true': 50
+  Goal:
+    three_proofs: 25
+    lucky_emblem_hunt: 25
+  FinalXemnas:
+    'true': 50
+  LuckyEmblemsAmount:
+    30: 50
```
Comment (Silvris) [inline games/Kingdom Hearts 2.yaml:31]:
```suggestion
    random-range-30-35: 50
```
can cut this down by 5 rows

**Review by Silvris — COMMENTED** [review]
Recommendations are just to make the yaml a little cleaner, settings are good otherwise.

**Comment by shananas** [comment]
changed Lucky Emblem stuff to use the range suggested by Silvris didn't know you could do that.

**Pull request merged**